acurl/acurl/__init__.py

```diff
@@ -400,13 +401,28 @@ def headers(self):
                 self._headers[k] = ", ".join(headers[k])
         return self._headers
 
+    def _get_header_lines(self):
+        header_lines = iter(self.header.splitlines())
+        for line in header_lines:
```

Comment:
This lets us have "100 continue" lines at any point in the response, right?  But they can only happen at the beginning, right?  It's also less efficient because it's doing a regex match against each line.

I'd rather we implemented this in a much simpler way:

```
headers = [l for l in self.headers.split("\r\n")]
headers = headers[:-2] # drop the final blank lines
if headers[0].startswith("HTTP/1.1 100"):  # http 1.0 doesn't define 100 continue; and we can be pretty sure there's no http/1.2 in the offing....
    headers = headers[2:]  # it's not clear to me if a 100 Continue response can have headers itself or what we'd do if it did
return headers
```

Reply:
So, I went with the iterative approach because I wasn't sure if you could encounter multiple HTTP 100s before getting the final response code. Even so, we could certainly take your approach and adapt it to cope with many, with the addition of a single line. But I'm not sure if the spec's particularly clear about the possibility. Will use your version as writted for now and we can worry about it when it (probably never) happens:

https://stackoverflow.com/questions/22818059/several-100-continue-received-from-the-server

Reply:
In fact it's not even a new line is it...

Reply:
No newline, just a carriage return 😝